include/shell/shell_dummy.h

```diff
@@ -17,6 +19,12 @@ extern const struct shell_transport_api shell_dummy_transport_api;
 
 struct shell_dummy {
 	bool initialized;
+
+	/** current position of output pointer (0 if no output) */
+	size_t ptr;
```

Comment:
imo this variable name is a bit misleading. I think that cnt, count, len, amount would be describe better this variable.

Reply:
OK I'll go with 'len' since cnt is already used in the shell's write() API and it gets a bit confusing